mllib/src/main/scala/org/apache/spark/mllib/stat/test/KSTest.scala

```diff
@@ -0,0 +1,191 @@
+   * @param cdf `Double => Double` function to calculate the theoretical CDF
+   * @return KSTestResult summarizing the test results (pval, statistic, and null hypothesis)
+   */
+  def testOneSample(data: RDD[Double], cdf: Double => Double): KSTestResult = {
+    val n = data.count().toDouble
+    val localData = data.sortBy(x => x).mapPartitions { part =>
+      val partDiffs = oneSampleDifferences(part, n, cdf) // local distances
+      searchOneSampleCandidates(partDiffs) // candidates: local extrema
+    }.collect()
+    val ksStat = searchOneSampleStatistic(localData, n) // result: global extreme
+    evalOneSampleP(ksStat, n.toLong)
+  }
+
+  /**
+   * Runs a KS test for 1 set of sample data, comparing it to a theoretical distribution
+   * @param data `RDD[Double]` data on which to run test
+   * @param createDist `Unit => RealDistribution` function to create a theoretical distribution
+   * @return KSTestResult summarizing the test results (pval, statistic, and null hypothesis)
+   */
+  def testOneSample(data: RDD[Double], createDist: () => RealDistribution): KSTestResult = {
+    val n = data.count().toDouble
+    val localData = data.sortBy(x => x).mapPartitions { part =>
+      val partDiffs = oneSampleDifferences(part, n, createDist) // local distances
+      searchOneSampleCandidates(partDiffs) // candidates: local extrema
+    }.collect()
+    val ksStat = searchOneSampleStatistic(localData, n) // result: global extreme
+    evalOneSampleP(ksStat, n.toLong)
+  }
+
+  /**
+   * Calculate unadjusted distances between the empirical CDF and the theoretical CDF in a
+   * partition
+   * @param partData `Iterator[Double]` 1 partition of a sorted RDD
+   * @param n `Double` the total size of the RDD
+   * @param cdf `Double => Double` a function the calculates the theoretical CDF of a value
+   * @return `Iterator[Double] `Unadjusted (ie. off by a constant) differences between
+   *        ECDF (empirical cumulative distribution function) and CDF. We subtract in such a way
+   *        that when adjusted by the appropriate constant, the difference will be equivalent
+   *        to the KS statistic calculation described in
+   *        http://www.itl.nist.gov/div898/handbook/eda/section3/eda35g.htm
+   *        where the difference is not exactly symmetric
+   */
+  private def oneSampleDifferences(partData: Iterator[Double], n: Double, cdf: Double => Double)
+    : Iterator[Double] = {
+    // zip data with index (within that partition)
+    // calculate local (unadjusted) ECDF and subtract CDF
+    partData.zipWithIndex.map { case (v, ix) =>
+      // dp and dl are later adjusted by constant, when global info is available
+      val dp = (ix + 1) / n
+      val dl = ix / n
+      val cdfVal = cdf(v)
+      // if dp > cdfVal the adjusted dp is still above cdfVal, if dp < cdfVal
+      // we want negative distance so that constant adjusted gives correct distance
+      if (dp > cdfVal) dp - cdfVal else dl - cdfVal
```

Comment:
Yes, you're absolutely right. I made this change, which also simplifies the searching of the global extrema, since we no longer have to adjust by 1/N. Also fixed an edge case for when a partition has no data.